### PR TITLE
Alessandro/web 1388 mutation create new email

### PR DESCRIPTION
**Pull request by alemagio**
## Description & motivation

- emails list query
- create user email mutation
- delete user email mutation
- set primary email mutation

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/server/modules/core/graph/resolvers/userEmails.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  createUserEmailFactory,
+  deleteUserEmailFactory,
+  findEmailsByUserIdFactory,
+  setPrimaryUserEmailFactory
+} from '@/modules/core/repositories/userEmails'
+import { db } from '@/db/knex'
+
+export = {
+  Query: {
+    userEmails: (_parent, _args, ctx) => {
```
Comment (alemagio) [inline packages/server/modules/core/graph/resolvers/userEmails.ts:12]: need to add pagination here

File: packages/server/modules/core/tests/graph/userEmails.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      const res = await sendRequest(token, {
+        query: `query { userEmails { id } }`
+      })
+
+      expect(res.error).to.not.ok
+      expect(res.body.data.userEmails).to.have.length(2)
+    })
+  })
+
+  describe('createUserEmail mutation', () => {
```
Comment (fabis94) [inline packages/server/modules/core/tests/graph/userEmails.spec.ts:51]: All of your test cases are only validating successful execution paths. I suggest using `itEach()` to run the same queries/mutations with different kinds of params, validating what happens with invalid inputs.

**Review by fabis94 — CHANGES_REQUESTED** [review]

File: packages/server/modules/core/graph/resolvers/userEmails.ts (hunk 1 above)
Comment (alemagio) [reply packages/server/modules/core/graph/resolvers/userEmails.ts:12]: yeah, agree, I think we can skip it for the moment

**Review by fabis94 — CHANGES_REQUESTED** [review]
I'm fine with the logic, just want the typedefs to be re-organized and more appropriate for how the client is gonna use them

**Review by fabis94 — DISMISSED** [review]

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**